lib/go/test/flow_idtable_staking_test.go

```diff
@@ -2303,3 +2303,246 @@ func TestIDTableStaking(t *testing.T) {
+		nodeRewardWithoutWithold := totalPayout.Div(totalStaked).Mul(amountToCommit).(interpreter.UFix64Value)
+		delegatorReward := totalPayout.Div(totalStaked).Mul(delegatorCommitment).(interpreter.UFix64Value)
+		delegatorRewardNodeCut := delegatorReward.Mul(cutPercentage).(interpreter.UFix64Value)
+		delegatorRewardMinusNode := delegatorReward.Minus(delegatorRewardNodeCut).(interpreter.UFix64Value)
+
+		// For multiplying the reward amounts
+		var numDelegators interpreter.UFix64Value = 500000000
+
+		// The rewards for a node and its 5 delegators
+		// without including withheld rewards from other nodes
+		nodeRewardPlusDelegators := nodeRewardWithoutWithold.Plus(delegatorRewardNodeCut.Mul(numDelegators)).(interpreter.UFix64Value)
+
+		// For multiplying the reward amounts
+		var numWithheldNodes interpreter.UFix64Value = 200000000
+
+		// Figure out the sum of tokens withheld from all punished nodes
+		amountWithheld := nodeRewardWithoutWithold.Mul(numWithheldNodes).Plus(delegatorReward.Mul(numDelegators)).(interpreter.UFix64Value)
+
+		// Calculate the additional tokens to give to nodes and delegators
+		// only from the withheld tokens
+		nodeRewardFromWithheld := amountWithheld.Div(totalStaked).Mul(amountToCommit).(interpreter.UFix64Value)
+		delegatorRewardFromWithheld := amountWithheld.Div(totalStaked).Mul(delegatorCommitment).(interpreter.UFix64Value)
+		delegatorRewardNodeCutFromWithheld := delegatorRewardFromWithheld.Mul(cutPercentage).(interpreter.UFix64Value)
+		delegatorRewardMinusNodeFromWithheld := delegatorRewardFromWithheld.Minus(delegatorRewardNodeCutFromWithheld).(interpreter.UFix64Value)
+
+		// Add the normal rewards to the rewards from withholding
+		totalNodeReward := nodeRewardWithoutWithold.Plus(nodeRewardFromWithheld).(interpreter.UFix64Value)
+		totalNodeRewardPlusDelegators := nodeRewardPlusDelegators.Plus(nodeRewardFromWithheld.Plus(delegatorRewardNodeCutFromWithheld.Mul(numDelegators))).(interpreter.UFix64Value)
+		totalDelegatorReward := delegatorRewardMinusNode.Plus(delegatorRewardMinusNodeFromWithheld).(interpreter.UFix64Value)
+
+		// Nodes 1, 3-9
+		for i := 1; i < numNodes; i++ {
+			if i == 1 {
+				result = executeScriptAndCheck(t, b, templates.GenerateGetRewardBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[i]))})
+				assertEqual(t, CadenceUFix64(totalNodeRewardPlusDelegators.String()), result)
+			} else if i != 2 {
+				result = executeScriptAndCheck(t, b, templates.GenerateGetRewardBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[i]))})
+				assertEqual(t, CadenceUFix64(totalNodeReward.String()), result)
+			}
+		}
+
+		// Delegators from Node 1
+		for i := 1; i < 6; i++ {
+			result = executeScriptAndCheck(t, b, templates.GenerateGetDelegatorRewardsScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(ids[1])), jsoncdc.MustEncode(cadence.UInt32(i))})
+			assertEqual(t, CadenceUFix64(totalDelegatorReward.String()), result)
+		}
+	})
```

Comment:
Could we add a test (or assertion to an existing) test which checks that the total amount of rewards is the same regardless of any withholdings? 

I'm not too familiar, but I found it a bit hard to convince myself of this reading through the reward calculation logic.

Reply:
> Could we add a test (or assertion to an existing) test which checks that the total amount of rewards is the same regardless of any withholdings?
➕

Reply:
Good catch, I added that and there was a missed calculation in there. The calculation and rewards paid are correct now